File: src/java/magma/api/JavaFiles.java
package magma.api;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.Optional;

public class JavaFiles {
    public static Optional<IOException> writeString(Path path, String content) {
        try {
            Files.writeString(path, content);
            return Optional.empty();
        } catch (IOException e) {
            return Optional.of(e);
        }
    }
}

File: src/c/magma/java/JavaFiles.c
#include "temp.h"
#include "temp.h"
#include "temp.h"
#include "temp.h"
#include "temp.h"
#include "temp.h"
#include "temp.h"
#include "temp.h"
struct JavaFiles {
	void temp(){
	to = from;catch (IOException e) {
            return new Err<>(e);
        }
	}
	void temp(){try {
            Files.createDirectories(targetParent);
            return Optional.empty();
        }catch (IOException e) {
            return Optional.of(e);
        }
	}
	void temp(){try {
            Files.writeString(target, output);
            return Optional.empty();
        }catch (IOException e) {
            return Optional.of(e);
        }
	}
	void temp(){try {
            return new Ok<>(Files.readString(source));
        }catch (IOException e) {
            return new Err<>(e);
        }
	}
};